src/revwalk.c

```diff
@@ -38,7 +38,7 @@ git_commit_list_node *git_revwalk__commit_lookup(
 	return commit;
 }
 
-static int push_commit(git_revwalk *walk, const git_oid *oid, int uninteresting, int from_glob)
+int git_revwalk__push_commit(git_revwalk *walk, const git_oid *oid, const git_revwalk__push_options *opts)
```

Comment:
We should probably fall back to default options if `opts == NULL`?

Reply:
I wanted the argument to be mandatory, mostly because the "mode" kinda is (OID vs glob, else it's ambiguous what should happen w.r.t to peeling). But I forgot to assert that though 😉.

Reply:
Fair enough, it's internal anyway so all callers can be checked trivially. Let's just go ahead